primedev/mods/modweaponvars.cpp

```diff
@@ -0,0 +1,103 @@
+
+#include "client/weaponx.h"
+#include "server/weaponx.h"
+#include "modweaponvars.h"
+#include "squirrel/squirrel.h"
+#include "client/r2client.h"
+#include "server/r2server.h"
+
+template <ScriptContext context> WeaponVarInfo* weaponVarArray;
+
+AUTOHOOK_INIT()
+
+// TODO: _actually_ reverse this.
+// Seemingly recalculates weapon variables every tick, figure out what this actually does
+// rn we are "disabling" this function because it seems to make stuff
+// work, figure out why.
+// also ""slight"" performance increase
+AUTOHOOK(Cl_UpdateWeaponVars, client.dll + 0x5B2670, char, __fastcall, (void* a2))
```

Comment:
Bit scary but sure

Reply:
A better solution is to probably hook the function that actually recalculates mod values and do a codecallback after it